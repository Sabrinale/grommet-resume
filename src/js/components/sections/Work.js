import React from 'react'
import { Box } from 'grommet'

import { formatDate } from '../common'
import { Description } from '../Description'
import { Highlights } from '../Highlights'
import { SectionTitle } from '../SectionTitle'
import { DateSection } from '../DateSection'
import { TagSection } from '../TagSection'

const WorkItem = ({ item }) => {
  const startDate = formatDate(item.startDate)
  const endDate = formatDate(item.endDate)

  console.log('Highlights: ', Highlights)
  console.log('SectionTitle: ', SectionTitle)
  console.log('Box: ', Box)
  console.log('DateSection: ', DateSection)
  console.log('Description: ', Description)
  console.log('TagSection: ', TagSection)

  return (
    <Highlights>
      <SectionTitle>{item.company}</SectionTitle>
      <Box direction="row">
        <DateSection>{`${startDate} - ${endDate}`}</DateSection>
      </Box>
      <Description>{item.summary}</Description>
      <TagSection tags={item.highlights}>Highlights</TagSection>
    </Highlights>
  )
}

export { WorkItem }
