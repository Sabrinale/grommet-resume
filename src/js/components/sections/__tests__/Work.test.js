import React from 'react'
import renderer from 'react-test-renderer'

import { WorkItem } from '../Work'

import defaultResume from './mockResume.json'

test('renders work section', () => {
  const data = defaultResume.work[0]
  const component = renderer.create(<WorkItem item={data} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
