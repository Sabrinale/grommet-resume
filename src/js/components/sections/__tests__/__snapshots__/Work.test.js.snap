// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders work section 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 kFdcfG"
>
  <h3
    className="StyledHeading-sc-1rdh4aw-0 jVEgcZ"
  >
    Company
  </h3>
  <div
    className="StyledBox-sc-13pk1d4-0 ffeAql"
  >
    <span
      className="StyledText-sc-1sadyjn-0 dndifk"
      size="medium"
    >
      Jan 2013 - Jan 2014
    </span>
  </div>
  <p
    className="StyledParagraph-tbetod-0 jzEULC"
  >
    Description...
  </p>
  <div
    className="StyledBox-sc-13pk1d4-0 IRSNj"
  >
    <span
      className="StyledText-sc-1sadyjn-0 dndifk"
      size="medium"
    >
      Highlights
    </span>
    <div
      className="StyledBox-sc-13pk1d4-0 dvQWRk"
    >
      <span
        className="StyledText-sc-1sadyjn-0 fWSbXS"
      >
        Started the company
      </span>
    </div>
  </div>
</div>
`;
